orchagent/vnetorch.h

```diff
@@ -283,10 +285,25 @@ struct VNetNextHopObserverEntry
 /* NextHopObserverTable: Destination IP address, next hop observer entry */
 typedef std::map<IpAddress, VNetNextHopObserverEntry> VNetNextHopObserverTable;
 
+struct VNetNextHopInfo
+{
+    IpAddress monitor_addr;
+    sai_bfd_session_state_t bfd_state;
+    std::set<NextHopGroupKey> nhgs;
+};
+
+struct BfdSessionInfo
+{
+    sai_bfd_session_state_t bfd_state;
+    std::map<std::string, NextHopKey> endpoints;
+};
+
 typedef std::map<NextHopGroupKey, NextHopGroupInfo> VNetNextHopGroupInfoTable;
 typedef std::map<IpPrefix, NextHopGroupKey> VNetTunnelRouteTable;
+typedef std::map<IpAddress, BfdSessionInfo> BfdSessionTable;
+typedef std::map<IpAddress, VNetNextHopInfo> VNetEndpointInfoTable;
```

Comment:
Lets discuss this map? I'm not fully understanding the purpose

Reply:
The purpose of this map is to track the BFD state and nexthop groups of each endpoint so that the nexthop group can be updated accordingly. I believe we do need some discussion about the implementation. Will get it scheduled.